service/resource/vaultpki/create.go

```diff
@@ -0,0 +1,70 @@
+package vaultpki
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/cert-operator/service/key"
+)
+
+func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	currentVaultPKIState, err := toVaultPKIState(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredVaultPKIState, err := toVaultPKIState(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out if the Vault PKI has to be created")
+
+	var vaultPKIStateToCreate VaultPKIState
+	if !currentVaultPKIState.BackendExists || !currentVaultPKIState.CAExists {
+		vaultPKIStateToCreate = desiredVaultPKIState
+	}
```

Comment:
As a first step we just work with a custom structure to reflect boolean states for the PKI backend and the root CA. We are still only interested in this binary states because we just want to make sure the PKI backend is setup or not. Later we can potentially extend this easily by tuning the PKI backend mount in Vault. This is out of scope of this PR though.